public/app/features/alerting/unified/components/silences/MatchedSilencedRules.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { useDebounce } from 'react-use';
+import { useDispatch } from 'react-redux';
+import { css } from '@emotion/css';
+import { GrafanaTheme2 } from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
+import { DynamicTable, DynamicTableColumnProps, DynamicTableItemProps } from '../DynamicTable';
+import { RuleState } from '../rules/RuleState';
+import { useCombinedRuleNamespaces } from '../../hooks/useCombinedRuleNamespaces';
+import { Annotation } from '../../utils/constants';
+import { findAlertRulesWithMatchers } from '../../utils/matchers';
+import { fetchAllPromAndRulerRulesAction } from '../../state/actions';
+import { CombinedRule } from 'app/types/unified-alerting';
+import { MatcherFieldValue, SilenceFormFields } from '../../types/silence-form';
+
+type MatchedRulesTableItemProps = DynamicTableItemProps<{
+  matchedRule: CombinedRule;
+}>;
+type MatchedRulesTableColumnProps = DynamicTableColumnProps<{ matchedRule: CombinedRule }>;
+
+export const MatchedSilencedRules = () => {
+  const [matchedAlertRules, setMatchedAlertRules] = useState<MatchedRulesTableItemProps[]>();
```

Comment:
If we set the default value to `[]` we can omit most of the `matchedAlertRules &&` and `matchedAlertRules?` checks